src/scipp/interpolate/__init__.py

```diff
@@ -78,6 +78,12 @@ class returns a function whose call method uses interpolation to find the value
     :py:class:`scipp.DimensionError` is raised since interpolation requires points to
     be 1-D.
 
+    For structured input data dtypes such as vectors, rotations, or linear
+    transformations interpolation is structure-element-wise. While this is appropriate
+    for vectors, such a naive interpolation for, e.g., rotations does typically not
+    yield a rotation so this should be used with care, unless the 'kind' parameter is
+    set to, e.g., 'previous', 'next', or 'nearest'.
+
```

Comment:
I wonder whether it is worth just throwing an error for rotation and affine3d types if `kind=linear` to prevent these sorts of issues in the first place? I can't think of a case where the current implementation with kind=linear would actually be useful for those types, and the resulting transform probably has undefined behaviour in many cases as your comment suggests (e.g. if the rotation isn't normalised). I guess I'm worried that someone thinks they can use this as an implementation of SLERP or similar.

Long-term we might want to specialise an implementation for rotations. However I'm not suggesting we need to do this work as part of this issue, or soon.

Reply:
Not sure. I think for *any* function there are inputs that do not entirely make sense, but restricting things too much will make the entire toolkit less useful. For example, computing `mean` of data with 'counts' unit is usually not "correct" (`sum` should be used)... except when someone is, e.g., studying faulty detectors.

Similarly:
- using `interp1d` on `affine_transform` is not correct... unless the users knows it only has a translation component.
- using `interp1d` on `linear_transform` is not correct... unless the user knows they have just a scaling.
- ...